### PR TITLE
Wcag/8 fix accessibiliy issues reported by eslint

### PR DESCRIPTION
**Pull request by mjulstein**
## Description
Focusing on the accessibility issues reported by eslint (a11y )
**Note:** These changes are not tested with other means than the unittests,
TODO:
 - I need to find or create a case where the components are in use in order to test if I broke anything.
 - I should create unit-tests once the components are verified.

## Fixes
- #8 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

### CODE REVIEW

File: src/shared/src/components/molecules/AltinnModal.tsx
Comment (haakemon) [inline src/shared/src/components/molecules/AltinnModal.tsx:110]: ~~This doesnt seem entirely correct to me - this will remove the ability to navigate to this by tabbing. I would think moving the tabIndex property to the parent `IconButton` would be better?~~ 

Or maybe its not needed at all, since I guess `IconButton` uses a `button` under the hood anyway, and so it will by default be possible to navigate to it.

So `tabindex` could probably removed entirely in this case.

WDYT?

File: src/shared/src/components/molecules/AltinnModal.tsx
Comment (haakemon) [inline src/shared/src/components/molecules/AltinnModal.tsx:135]: I would suggest here as well to destruct immediately:

```suggestion
const AltinnModal = ({isOpen, hideBackdrop, allowCloseOnBackdropClick, onClose, printView, ...rest}: React.PropsWithChildren<IAltinnModalComponentProvidedProps>) => {
  const classes = useStyles();
  if (!printView) {
    return (
      <Modal
        open={isOpen}
        className={classes.scroll}
        hideBackdrop={hideBackdrop}
        onBackdropClick={allowCloseOnBackdropClick === false ? null : onClose}
      >
        <StyledModal onClose={onClose} {...rest}/>
```

File: src/shared/src/components/AltinnInput.tsx
Comment (haakemon) [inline src/shared/src/components/AltinnInput.tsx:96]: Do we not need the click handler here to focus the input?

File: src/altinn-app-frontend/src/components/summary/SummaryComponent.tsx
Comment (haakemon) [inline src/altinn-app-frontend/src/components/summary/SummaryComponent.tsx:164]: Not sure we can change the casings here - this is case sensitive, isnt it?

File: src/shared/src/components/molecules/AltinnModal.tsx
Comment (mjulstein) [reply src/shared/src/components/molecules/AltinnModal.tsx:110]: I think you are right. My first intent was to remove ( see first commit ), but because I don't currently know how this affects the component in practice I wanted a second opinion. From what I could get by asking @lorang92 yesterday, the other case where `tabindex={0}` was used it was intentionally because of the need to use `focus()` on an non interactive element.
In this case I'm not sure, but left it if that was the case here as well.
WDYT?

File: src/shared/src/components/AltinnInput.tsx
Comment (mjulstein) [reply src/shared/src/components/AltinnInput.tsx:96]: My understanding was that it was redundant as it is also on the surrounding grid on line 74?

File: src/altinn-app-frontend/src/components/base/InputComponent.tsx
Comment (haakemon) [reply src/altinn-app-frontend/src/components/base/InputComponent.tsx:70]: Ah, you are right. Sorry for the confusion :) 

The ID is still part of the `props` as you mention, so it is not lost.

File: src/shared/src/components/AltinnInput.tsx
Comment (haakemon) [reply src/shared/src/components/AltinnInput.tsx:96]: Ah, that part was not visible by default in the PR changes tab, so didnt notice that. Could very well be that this is covered by the grid parent.

File: src/altinn-app-frontend/src/components/summary/SummaryComponent.tsx
Comment (mjulstein) [reply src/altinn-app-frontend/src/components/summary/SummaryComponent.tsx:164]: On line 155 i set the formComponent.type.toLowerCase().
I also removed case `Group` as it seems that having a case for every individual casing possible seems a bit messy to me.

File: src/altinn-app-frontend/src/components/base/InputComponent.tsx
Comment (mjulstein) [reply src/altinn-app-frontend/src/components/base/InputComponent.tsx]: I realized where you were coming from, and had to reference the id in order to have the destructor in the args.
Could probably rename the rest to something more fitting to the intent ie. htmlProps, but I think these problems are more important to cover when porting over to the design system.

File: src/shared/src/components/molecules/AltinnModal.tsx
Comment (mjulstein) [reply src/shared/src/components/molecules/AltinnModal.tsx]: In this case I'm not sure if I want to move so much at this point.
I will then have write the unit-tests first, so I know that all args are properly included in both branches.

File: src/altinn-app-frontend/src/components/summary/SummaryComponent.tsx
Comment (haakemon) [reply src/altinn-app-frontend/src/components/summary/SummaryComponent.tsx:173]: Aha, I missed that part... But is it a good idea to change this? Then it will be out of sync with casing from formlayout.

An even better approach would be to use an enum for these instead I think. Then we would also get an error if something is misspelled.

File: src/shared/src/components/molecules/AltinnModal.tsx
Comment (haakemon) [reply src/shared/src/components/molecules/AltinnModal.tsx]: We also know what props `StyledModal` needs. But I agree its not really clear - this is one of the downsides of just spreading props like this :) 

Adding more tests to this is a good idea though, and that would make refactoring safer.

File: src/altinn-app-frontend/src/components/base/InputComponent.tsx
Comment (haakemon) [reply src/altinn-app-frontend/src/components/base/InputComponent.tsx]: I dont agree with that argument :) Why not change it now, and clean up the code immediately? Whats the reason to wait until this is ported to the design system? Then we have to remember to fix it then instead.

File: src/altinn-app-frontend/src/components/summary/InputSummaryBoilerplate.tsx
Comment (haakemon) [inline src/altinn-app-frontend/src/components/summary/InputSummaryBoilerplate.tsx:8]: This approach works fine, but I wanted to mention custom hooks as an alternative 🙂
(I wrote the hook directly here, so its not tested and may typos or other issues - so if you change to this, be sure to check that it works as expected 😅)

```jsx
const useDisplayData = ({formData}) => {
  const [displayData, setDisplayData] = React.useState('');

  React.useEffect(() => {
    if (formData && typeof formData === 'object') {
      let displayString = '';
      Object.keys(formData).forEach((key, index) => {
        displayString += `${index > 0 ? ' ' : ''}${formData[key]}`;
      });
      setDisplayData(displayString);
    } else {
      setDisplayData(formData);
    }
  }, [formData, setDisplayData]);

  return {displayData}
}
```

And usage would be
```jsx

function GroupInputSummary({ formData, label }: ISingleInputSummary) {
  const classes = useStyles();

  const {displayData} = useDisplayData({formData});

  return (
      <Typography variant='body1'>
        <span className={classes.label}>
          {label} {': '}
        </span>
        <span>{displayData}</span>
      </Typography>
    )
}
```

Custom hooks is a nice way to reuse logic across components that doesnt have rendered output. We avoid the wrapping components and having to pass down extra props or state handlers etc..

File: src/altinn-app-frontend/src/components/summary/SummaryWrapper.test.tsx
Comment (haakemon) [inline src/altinn-app-frontend/src/components/summary/SummaryWrapper.test.tsx:20]: Isnt
```
expect(screen.queryByRole('button')).not.toBeNull();
```
and 
```
expect(screen.getByRole('button')).toBeInTheDocument();
```
the same thing?

File: src/altinn-app-frontend/src/components/summary/InputSummaryBoilerplate.tsx
Comment (mjulstein) [reply src/altinn-app-frontend/src/components/summary/InputSummaryBoilerplate.tsx:8]: Nice

File: src/altinn-app-frontend/src/components/summary/SummaryWrapper.test.tsx
Comment (mjulstein) [reply src/altinn-app-frontend/src/components/summary/SummaryWrapper.test.tsx:20]: Indeed they are checking for the same, I was thinking more in the lines of the other tests that use the queryByRole to verify the opposite. I guess line 20 would fail if 18 wasn't implicitly passed so I can take it away.

File: src/altinn-app-frontend/src/components/hooks/useDisplayData.ts
Comment (haakemon) [inline src/altinn-app-frontend/src/components/hooks/useDisplayData.ts:3]: We should try to not introduce new `any` types - I think we have typings for this already (`IFormData`), can you fix? 🙂

**Review by haakemon — APPROVED** [review]
This task grew quite quickly from the small fix it was supposed to be 😅

But really great to get this cleaned up! Great work 🥇

**Comment by haakemon** [comment]
Dont forget to bump version in package.json before merge

- [ ] bump package.json

**Pull request merged**
